crawler.py

```diff
@@ -38,6 +38,9 @@
 URL_PRINT_LENGTH = 300  # print at maximimum n chars of the url in error messages
 KNOWN_FILE = "known.txt"
 
+### HTTP request settings
+HEADERS = {'User-Agent': 'Mozilla/5.0 (X11; Linux x86_64; rv:74.0) Gecko/20100101 Firefox/74.0'} # set user agent to prevent "403: forbidden" errors
```

Comment:
This will not pass through [`black`](https://black.readthedocs.io/en/stable/).
  * the line's too long (88 chars or less)
  * 2 spaces before the comment
  * double quotes

My suggestion:

```suggestion
HEADERS = {
    # Set a user agent to prevent "403: forbidden" errors on some sites.
    "User-Agent": "Mozilla/5.0 (X11; Linux x86_64; rv:74.0) Gecko/20100101 Firefox/74.0"
}
```

Reply:
(I have `black` enabled on-save in my editor, for python files in my own projects -- very handy :smile: )

Reply:
black is nice, thanks a lot for this tool tip!